File: array/matrix_3x3.py
# 2. 두개의 3X3 2차원 배열을 입력 받아서,
#     1) 두 행렬의 합
#     2) 두 행렬의 차
#     3) 두 행렬의 곱
# 연산을 선택하여 출력하는 프로그램을  구현해보세요.

def metrix_sum(m1, m2):
    result = []
    for i in range(9):
        result.append(m1[i]+m2[i])
    return result

def metrix_minus(m1, m2):
    result = []
    for i in range(9):
        result.append(m1[i]-m2[i])
    return result

def metrix_mul(m1, m2):
    result = []
    for i in range(3):
        for j in range(3):
            tmp = m1[3*i] * m2[j]
            tmp += m1[3*i+1] * m2[3+j]
            tmp += m1[3*i+2] * m2[6+j]
            result.append(tmp)
    return result

def print_metrix(m):
    for i in range(3):
        for j in range(3):
            print("{} ".format(m[3*i+j]), end='')
        print()


m1 = []
m2 = []
for i in range(3):
    for j in range(3):
        m1.append(int(input('첫번째 행렬에 넣을 숫자를 입력하세요 : ')))

for i in range(3):
    for j in range(3):
        m2.append(int(input('두번째 행렬에 넣을 숫자를 입력하세요 : ')))

print('첫번째 행렬')
print_metrix(m1)
print()
print('두번째 행렬')
print_metrix(m2)

while True:
    print('실행할 계산을 선택해주세요')
    print('1. 행렬의 덧셈')
    print('2. 행렬의 뺄셈')
    print('3. 행렬의 곱셈')
    print('(나머지 값 입력시 종료)')
    choice = int(input(' >> '))
    if choice == 1:
        print('덧셈 결과')
        print_metrix(metrix_sum(m1, m2))
    elif choice == 2:
        print('뺼셈 결과')
        print_metrix(metrix_minus(m1, m2))
    elif choice == 3:
        print('곱셈 결과')
        print_metrix(metrix_mul(m1, m2))
    else:
        print('종료')
        break
input('')

File: array/matrix_3x3.cpp
//# 2. 두개의 3X3 2차원 배열을 입력 받아서,
//#     1) 두 행렬의 합
//#     2) 두 행렬의 차
//#     3) 두 행렬의 곱
//# 연산을 선택하여 출력하는 프로그램을  구현해보세요.
#include <stdio.h>
#include <stdlib.h>
#define LENGTH 9

void print_metrix(int *m){
    for(int i=0;i<3;i++){
        for(int j=0;j<3;j++) printf("%d ", m[3*i+j]);
        printf("\n");
    }
}

void metrix_sum(int *r, int *m1, int *m2){
    for(int i=0;i<LENGTH;i++) *(r+i) = *(m1+i) + *(m2+i);
}
void metrix_minus(int *r, int *m1, int *m2){
    for(int i=0;i<LENGTH;i++) *(r+i) = *(m1+i) - *(m2+i);
}
void metrix_mul(int *r, int *m1, int *m2){
    int tmp;
    for(int i=0;i<3;i++){
        for(int j=0;j<3;j++){
            tmp=0;
            for(int k=0;k<3;k++) tmp += *(m1+3*i+k) * *(m2+3*k+j);
            *(r+i*3+j) = tmp;
        }
    }
}

int main(){
    int *a = (int *)malloc(sizeof(int)*LENGTH);
    int *b = (int *)malloc(sizeof(int)*LENGTH);
    printf("첫번째 3x3 행렬을 입력해주세요.\n");
    for(int i=0;i<LENGTH;i++) scanf("%d", a+i);
    printf("두번째 3x3 행렬을 입력해주세요.\n");
    for(int i=0;i<LENGTH;i++) scanf("%d", b+i);

    printf("<첫번째 배열입니다>\n");
    print_metrix(a);
    printf("\n<두번째 배열입니다>\n");
    print_metrix(b);

    int choice;
    int *result = (int *)malloc(sizeof(int)*9);
    while(1){
        printf("실행할 계산을 선택해주세요\n");
        printf("1. 행렬의 덧셈\n");
        printf("2. 행렬의 뺄셈\n");
        printf("3. 행렬의 곱셈\n");
        printf("(나머지 값 입력시 종료)\n");
        printf(" >> ");
        scanf("%d", &choice);
        if(choice==1){
            metrix_sum(result, a, b);
        } else if (choice == 2){
            metrix_minus(result, a, b);
        } else if (choice == 3){
            metrix_mul(result, a, b);
        } else break;
        printf("\n계산 결과입니다.\n");
        print_metrix(result);
    }
    free(a);
    free(b);
    free(result);
    getchar();
    getchar();
    return 0;
}